EXAMPLES.md

```diff
@@ -197,3 +197,121 @@ class AcsRespondController
+setup as of Symfony 5 made it much easier to facilitate this feature via the Stepup Saml Bundle.
+
+To get this feature to work you will need to perform the following steps
+
+1. Configure your project as an SP and set up at least one remote IdP. See steps above on some help with setting this up, or consult the README.md
+2. Create a `Security\Authentication\Provider\SamlProvider` This should implement the `Surfnet\SamlBundle\Security\Authentication\Provider\SamlProviderInterface` and the `UserProviderInterface`
+3. In `config/packages/security.yaml`: setup a `saml_based` firewall and create a `providers` entry. See code example 1 below.
+4. Add the `SamlProvider` service to your services.yaml. See example below in block 2.
+5. Configure the ACS route name of SP in your `.env` file. Example:  `acs_location_route_name='assertion_consumer_service'`. Your Saml Controller ACS location action should use the same route name. 
+6. Optionally create a `Security\Authentication\Handler\FailureHandler`. Here you can define what behavior is required when authentication failed. For example you can do a redirect to a certain page. Or show an error page. Must implement the `AuthenticationFailureHandlerInterface`. See code example 3 for details. A very simple implementation is provided in the bundle. Showing a very simple unstyled authn failed error page.
+ 
+### Code example 1: setting up security.yaml
+
+```yaml
+security:
+  # Other security entries are not shown in this code example. Only the required entries for getting SAML 
+  # authentications with the stepup-saml-bundle are stated below
+
+  access_control:
+    - { path: ^/saml, roles: IS_AUTHENTICATED_ANONYMOUSLY, requires_channel: https }
+    - { path: ^/, roles: IS_AUTHENTICATED_FULLY, requires_channel: https }
+  
+  providers:
+    saml-provider:
+      id: YourApp\Security\Authentication\Provider\SamlProvider
+      
+  firewalls:
+    login_firewall:
+      pattern:    ^/saml/metadata
+    
+    saml_based:
+      custom_authenticators:
+        - Surfnet\SamlBundle\Security\Authentication\SamlAuthenticator
+```
+
+### Code example 2: services.yaml
+```yaml
+
+    # This is just an example of the dependencies a SamlProvider could have, taylor this provider to your own needs
+    surfnet_saml.saml_provider:
+        class: YourApp\Security\Authentication\Provider\SamlProvider
+        arguments:
+            - '@YourApp\Repository\UserRepository'
+            - '@surfnet_saml.saml.attribute_dictionary'
+            - '@logger'
+
+    # Be sure to create an alias to the `surfnet_saml.saml_provider`, otherwise Symfony will not be able to autowire the
+    # `security.listener.user_provider` service.
+    Surfnet\ServiceProviderDashboard\Infrastructure\DashboardSamlBundle\Security\Authentication\Provider\SamlProvider:
+        alias: surfnet_saml.saml_provider
+
+```
+
+### Code example 3: example failure handler
+```php
+<?php
+
+/**
+ * Copyright 2023 SURFnet B.V.
```

Comment:
Is it really necessary to add a copyright notice to an example?  It doesn't really add to the readability

Reply:
Good point, removed that!